### ISSUE TITLE
D2: add dedicated evidence that internal custom-attribute decode resource exhaustion propagates

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#5397]
## Observed

`AttributeDecoder.TryDecode` ends with a bare catch (`AttributeDecoder.cs:305-308`):

```csharp
try
{
    return attribute.DecodeValue(provider);
}
catch (MaterializationObserverException ex)
{
    ex.Rethrow();
    throw;
}
catch
{
    return null;
}
```

The final `catch` has no filter, so it absorbs **every** exception type, including `OutOfMemoryException`. The guard's earlier `catch` is correctly filtered to `BadImageFormatException or ArgumentOutOfRangeException`; this one is not.

## Evidence

Reproduced at PR #5355's head while validating the classification canary, using the captured `dotnet/runtime#57531` blob and a provider mutated to misclassify the `System.Type` argument (the gap 8 / #5393 divergence). Run with `DOTNET_GCHeapHardLimit=40000000`:

```
PROBE guard=True srm=OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
```

The guard approves, `DecodeValue` requests roughly 28,515 MiB and throws `OutOfMemoryException` — and `TryDecode` returns `null`. The caller cannot distinguish "this attribute is not decodable" from "we just attempted a 28 GB allocation and the process may be in a degraded state."

## Why it matters

Two distinct problems:

1. **Failure is not visible.** The repository rule is to keep decode failures visible rather than turning them into success-shaped empty output. A swallowed `OutOfMemoryException` renders as a silently missing attribute.
2. **`OutOfMemoryException` is not a routine parse failure.** After a multi-gigabyte allocation attempt, continuing as though the blob were merely malformed is not a safe assumption, and the surrounding process may already be affected.

Note this is *not* the guard failing. The guard is doing its job; this is about what happens on the path the guard cannot cover today, which is exactly the classification divergence recorded as gap 8.

## Resolution direction

Filter the final catch to the exception types that genuinely mean "this blob is not decodable" (`BadImageFormatException`, `ArgumentOutOfRangeException`, and any others deliberately intended), and let the rest propagate. Consider whether the outcome should be a typed failure rather than `null`.

Related: #5393 (classification not shared), #5085 (gap 4, observer exception inverted into approval).

**Comment by richlander** [comment]
Reconciled under #5288 after #5815 merged: this issue remains OPEN. The broad catch is gone, and the owned decoder catches only malformed-structure exceptions; internal OutOfMemoryException now propagates by that implementation. Dedicated internal-exhaustion propagation evidence is still absent. Observer/resolver callback tests do not establish it, so full D2 remains partially unverified. The original bare-catch diagnosis below is historical; the remaining task is the owner-required internal-origin evidence, not another catch rewrite.

**Comment by richlander** [comment]
Focused evidence is proposed in #6123, head `ecbf16d25bee31ed3c668fc8dc1f569bb6bc21dc`. It uses SRM’s existing string-decoder customization point to inject one raw OutOfMemoryException beneath the public attribute decoder, with no observer or enum resolver. All three public decode surfaces preserve exception identity. A temporary refusal-policy mutation failed exactly those three cases while 110 neighboring cases passed; that mutation was removed and product code is unchanged. The claim is deterministic fault-injected propagation, not physical heap exhaustion, recovery under memory pressure, or exhaustive D2 coverage. This addresses the remaining evidence gap rather than rewriting the already-correct catch policy.
